src/cascadia/TerminalApp/KeyMapping.idl

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import "ActionArgs.idl";
+
+namespace TerminalApp
+{
+    enum ShortcutAction
+    {
+        Invalid = 0,
+        CopyText,
+        PasteText,
+        OpenNewTabDropdown,
+        DuplicateTab,
+        NewTab,
+        NewWindow,
+        CloseWindow,
+        CloseTab,
+        ClosePane,
+        NextTab,
+        PrevTab,
+        SplitVertical,
+        SplitHorizontal,
+        SendInput,
+        SplitPane,
+        TogglePaneZoom,
+        SwitchToTab,
+        AdjustFontSize,
+        ResetFontSize,
+        ScrollUp,
+        ScrollDown,
+        ScrollUpPage,
+        ScrollDownPage,
+        ResizePane,
+        MoveFocus,
+        Find,
+        ToggleRetroEffect,
+        ToggleFocusMode,
+        ToggleFullscreen,
+        ToggleAlwaysOnTop,
+        OpenSettings,
+        SetColorScheme,
+        SetTabColor,
+        OpenTabColorPicker,
+        RenameTab,
+        ExecuteCommandline,
+        ToggleCommandPalette,
+        CloseOtherTabs,
+        CloseTabsAfter,
+        TabSearch
+    };
+
+    [default_interface] runtimeclass ActionAndArgs {
+        ActionAndArgs();
+        IActionArgs Args;
+        ShortcutAction Action;
+    };
+
+    [default_interface] runtimeclass KeyMapping
+    {
+        KeyMapping();

```

Comment:
i'm going to ask this question every time I see a constructor here

**should an external consumer of this library be able to construct an empty keymapping**

Reply:
if not, remove this _and the factory_.